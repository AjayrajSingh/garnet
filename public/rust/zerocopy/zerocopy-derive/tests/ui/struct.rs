// Copyright 2019 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#[macro_use]
extern crate zerocopy;

fn main() {}

//
// Generic errors
//

#[derive(FromBytes)]
#[repr("foo")]
struct Generic1;

#[derive(FromBytes)]
#[repr(foo)]
struct Generic2;

#[derive(FromBytes)]
#[repr(u8)]
struct Generic3;

#[derive(FromBytes)]
#[repr(i8)]
struct Generic4;

#[derive(FromBytes)]
#[repr(u16)]
struct Generic5;

#[derive(FromBytes)]
#[repr(i16)]
struct Generic6;

#[derive(FromBytes)]
#[repr(u32)]
struct Generic7;

#[derive(FromBytes)]
#[repr(i32)]
struct Generic8;

#[derive(FromBytes)]
#[repr(u64)]
struct Generic9;

#[derive(FromBytes)]
#[repr(i64)]
struct Generic10;

#[derive(FromBytes)]
#[repr(usize)]
struct Generic11;

#[derive(FromBytes)]
#[repr(isize)]
struct Generic12;

#[derive(FromBytes)]
#[repr(transparent, packed)]
struct Generic14;

#[derive(FromBytes)]
#[repr(C, transparent)]
struct Generic15;

#[derive(FromBytes)]
struct Generic16;

//
// FromBytes errors
//

#[derive(FromBytes)]
#[repr(packed)]
struct FromBytes1;

//
// AsBytes errors
//

#[derive(AsBytes)]
#[repr(C)]
struct AsBytes1<T>(T);

//
// Unaligned errors
//

#[derive(Unaligned)]
#[repr(C, align(2))]
struct Unaligned1;

#[derive(Unaligned)]
#[repr(transparent, align(2))]
struct Unaligned2 {
    foo: u8,
}

#[derive(Unaligned)]
#[repr(packed, align(2))]
struct Unaligned3;
