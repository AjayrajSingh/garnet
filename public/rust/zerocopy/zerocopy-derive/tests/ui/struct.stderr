error: unrecognized representation hint
  --> $DIR/struct.rs:15:8
   |
15 | #[repr("foo")]
   |        ^^^^^

error: unrecognized representation hint
  --> $DIR/struct.rs:19:8
   |
19 | #[repr(foo)]
   |        ^^^

error: unsupported representation for deriving FromBytes, AsBytes, or Unaligned on a struct
  --> $DIR/struct.rs:23:8
   |
23 | #[repr(u8)]
   |        ^^

error: unsupported representation for deriving FromBytes, AsBytes, or Unaligned on a struct
  --> $DIR/struct.rs:27:8
   |
27 | #[repr(i8)]
   |        ^^

error: unsupported representation for deriving FromBytes, AsBytes, or Unaligned on a struct
  --> $DIR/struct.rs:31:8
   |
31 | #[repr(u16)]
   |        ^^^

error: unsupported representation for deriving FromBytes, AsBytes, or Unaligned on a struct
  --> $DIR/struct.rs:35:8
   |
35 | #[repr(i16)]
   |        ^^^

error: unsupported representation for deriving FromBytes, AsBytes, or Unaligned on a struct
  --> $DIR/struct.rs:39:8
   |
39 | #[repr(u32)]
   |        ^^^

error: unsupported representation for deriving FromBytes, AsBytes, or Unaligned on a struct
  --> $DIR/struct.rs:43:8
   |
43 | #[repr(i32)]
   |        ^^^

error: unsupported representation for deriving FromBytes, AsBytes, or Unaligned on a struct
  --> $DIR/struct.rs:47:8
   |
47 | #[repr(u64)]
   |        ^^^

error: unsupported representation for deriving FromBytes, AsBytes, or Unaligned on a struct
  --> $DIR/struct.rs:51:8
   |
51 | #[repr(i64)]
   |        ^^^

error: unsupported representation for deriving FromBytes, AsBytes, or Unaligned on a struct
  --> $DIR/struct.rs:55:8
   |
55 | #[repr(usize)]
   |        ^^^^^

error: unsupported representation for deriving FromBytes, AsBytes, or Unaligned on a struct
  --> $DIR/struct.rs:59:8
   |
59 | #[repr(isize)]
   |        ^^^^^

error: conflicting representation hints
  --> $DIR/struct.rs:63:1
   |
63 | / #[repr(transparent, packed)]
64 | | struct Generic14;
   | |_________________^

error: conflicting representation hints
  --> $DIR/struct.rs:67:1
   |
67 | / #[repr(C, transparent)]
68 | | struct Generic15;
   | |_________________^

error: must have a non-align #[repr(...)] attribute in order to guarantee this type's memory layout
  --> $DIR/struct.rs:70:10
   |
70 | #[derive(FromBytes)]
   |          ^^^^^^^^^

error: unsupported on types with type parameters
  --> $DIR/struct.rs:85:10
   |
85 | #[derive(AsBytes)]
   |          ^^^^^^^

error: cannot derive Unaligned with repr(align(N > 1))
  --> $DIR/struct.rs:94:1
   |
94 | / #[repr(C, align(2))]
95 | | struct Unaligned1;
   | |__________________^

error: cannot derive Unaligned with repr(align(N > 1))
   --> $DIR/struct.rs:98:1
    |
98  | / #[repr(transparent, align(2))]
99  | | struct Unaligned2 {
100 | |     foo: u8,
101 | | }
    | |_^

error: cannot derive Unaligned with repr(align(N > 1))
   --> $DIR/struct.rs:104:1
    |
104 | / #[repr(packed, align(2))]
105 | | struct Unaligned3;
    | |__________________^

error[E0565]: meta item in `repr` must be an identifier
  --> $DIR/struct.rs:15:8
   |
15 | #[repr("foo")]
   |        ^^^^^

error[E0552]: unrecognized representation hint
  --> $DIR/struct.rs:19:8
   |
19 | #[repr(foo)]
   |        ^^^

error[E0517]: attribute should be applied to enum
  --> $DIR/struct.rs:23:8
   |
23 | #[repr(u8)]
   |        ^^
24 | struct Generic3;
   | ---------------- not an enum

error[E0517]: attribute should be applied to enum
  --> $DIR/struct.rs:27:8
   |
27 | #[repr(i8)]
   |        ^^
28 | struct Generic4;
   | ---------------- not an enum

error[E0517]: attribute should be applied to enum
  --> $DIR/struct.rs:31:8
   |
31 | #[repr(u16)]
   |        ^^^
32 | struct Generic5;
   | ---------------- not an enum

error[E0517]: attribute should be applied to enum
  --> $DIR/struct.rs:35:8
   |
35 | #[repr(i16)]
   |        ^^^
36 | struct Generic6;
   | ---------------- not an enum

error[E0517]: attribute should be applied to enum
  --> $DIR/struct.rs:39:8
   |
39 | #[repr(u32)]
   |        ^^^
40 | struct Generic7;
   | ---------------- not an enum

error[E0517]: attribute should be applied to enum
  --> $DIR/struct.rs:43:8
   |
43 | #[repr(i32)]
   |        ^^^
44 | struct Generic8;
   | ---------------- not an enum

error[E0517]: attribute should be applied to enum
  --> $DIR/struct.rs:47:8
   |
47 | #[repr(u64)]
   |        ^^^
48 | struct Generic9;
   | ---------------- not an enum

error[E0517]: attribute should be applied to enum
  --> $DIR/struct.rs:51:8
   |
51 | #[repr(i64)]
   |        ^^^
52 | struct Generic10;
   | ----------------- not an enum

error[E0517]: attribute should be applied to enum
  --> $DIR/struct.rs:55:8
   |
55 | #[repr(usize)]
   |        ^^^^^
56 | struct Generic11;
   | ----------------- not an enum

error[E0517]: attribute should be applied to enum
  --> $DIR/struct.rs:59:8
   |
59 | #[repr(isize)]
   |        ^^^^^
60 | struct Generic12;
   | ----------------- not an enum

error[E0692]: transparent struct cannot have other repr hints
  --> $DIR/struct.rs:63:8
   |
63 | #[repr(transparent, packed)]
   |        ^^^^^^^^^^^  ^^^^^^

error[E0692]: transparent struct cannot have other repr hints
  --> $DIR/struct.rs:67:8
   |
67 | #[repr(C, transparent)]
   |        ^  ^^^^^^^^^^^

error[E0692]: transparent struct cannot have other repr hints
  --> $DIR/struct.rs:98:8
   |
98 | #[repr(transparent, align(2))]
   |        ^^^^^^^^^^^  ^^^^^^^^

error[E0690]: transparent struct needs exactly one non-zero-sized field, but has 0
  --> $DIR/struct.rs:64:1
   |
64 | struct Generic14;
   | ^^^^^^^^^^^^^^^^^
   |
   = note: non-zero-sized field

error[E0690]: transparent struct needs exactly one non-zero-sized field, but has 0
  --> $DIR/struct.rs:68:1
   |
68 | struct Generic15;
   | ^^^^^^^^^^^^^^^^^
   |
   = note: non-zero-sized field

error[E0587]: type has conflicting packed and align representation hints
   --> $DIR/struct.rs:105:1
    |
105 | struct Unaligned3;
    | ^^^^^^^^^^^^^^^^^^

error: aborting due to 37 previous errors

Some errors occurred: E0517, E0552, E0565, E0587, E0690, E0692.
For more information about an error, try `rustc --explain E0517`.
